cmake: openamp: Fix OpenAMP sample

SebastianBoe opened the pull request:
Stop specifying that the OpenAMP 'remote' app should use the same
board as the root app.

Also, add assertions to make sure that the user does not try to
override the board's that are specified in the app build scripts.

This fixes #10345

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
cmake: openamp: Fix OpenAMP sample

Stop specifying that the OpenAMP 'remote' app should use the same
board as the root app.

Also, add assertions to make sure that the user does not try to
override the board's that are specified in the app build scripts.

This fixes #10345

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

galak reviewed (commented):
Do we need same changes for samples/subsys/ipc/ipm_mcux?

SebastianBoe committed:
cmake: ipc: Fix ipm_mcux sample

Stop specifying that the 'remote' app should use the same board as the
root app.

Also, add assertions to make sure that the user does not try to
override the board that is specified in the app build scripts.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe commented:
> Do we need same changes for samples/subsys/ipc/ipm_mcux?

Yes, fixed.
